src/EnergyPlus/HybridModel.cc

```diff
@@ -142,29 +166,221 @@ namespace HybridModel {
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+                        }
+                    }
+
+					// Scenario 3: Three unknown parameters to solve
+					if (FlagHybridModel_TM && FlagHybridModel_AI && FlagHybridModel_PC) {
+						if (TemperatureSchPtr > 0 && HumidityRatioSchPtr > 0 && CO2ConcentrationSchPtr > 0) {
+                            HybridModelZone(ZonePtr).InternalThermalMassCalc_T = false;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_T = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_H = true;
+                            HybridModelZone(ZonePtr).InfiltrationCalc_C = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_T = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_H = true;
+                            HybridModelZone(ZonePtr).PeopelCountCalc_C = true;
+						}
+					}
+
+                    // Summarise hybridmodel flags
+                    if (HybridModelZone(ZonePtr).InternalThermalMassCalc_T){
+                        helper_InternalThermalMassCalc = true;
+                    }
+                    if (HybridModelZone(ZonePtr).InfiltrationCalc_T ||
+                        HybridModelZone(ZonePtr).InfiltrationCalc_H ||
+                        HybridModelZone(ZonePtr).InfiltrationCalc_C) {
+                        helper_AirInfiltrationCalc = true;
+                    }
+                    if (HybridModelZone(ZonePtr).PeopelCountCalc_T ||
+                        HybridModelZone(ZonePtr).PeopelCountCalc_H ||
+                        HybridModelZone(ZonePtr).PeopelCountCalc_C) {
+                        helper_PeopleCountCalc = true;
+                    }
+
+                    // Decide if system supply terms are sufficient to be included in the inverse solution
+                    // For now, the inverse algorithms can only solve one unkown parameters at a time.
+                    if ((SupplyAirTemperatureSchPtr > 0 && SupplyAirMassFlowRateSchPtr > 0) ||
+                        (SupplyAirHumidityRatioSchPtr > 0 && SupplyAirMassFlowRateSchPtr > 0) ||
+                        (SupplyAirCO2ConcentrationSchPtr > 0 && SupplyAirMassFlowRateSchPtr > 0)){
+                        HybridModelZone(ZonePtr).IncludeSystemSupplyParameters = true;
+                        helper_IncludeSystemSupply = true;
+                    }
+
+                    // Need to update this section when writing the code to solve 2 or more variables at the same time
+                    if (HybridModelZone(ZonePtr).InternalThermalMassCalc_T && HybridModelZone(ZonePtr).InfiltrationCalc_T) {
+                        HybridModelZone(ZonePtr).InfiltrationCalc_T = false;
```

Comment:
This should be right after user input - if FlagHybridModel_TM && FlagHybridModel_AI, raise error. Nothing to do with what schedule is defined.

Reply:
@tsbyq